src/coord/Axis.ts

```diff
@@ -246,7 +247,13 @@ class Axis {
         const axisExtent = this._extent;
         const dataExtent = this.scale.getExtent();
 
-        let len = dataExtent[1] - dataExtent[0] + (this.onBand ? 1 : 0);
+        let len = 0;
+        if (this.scale.type === 'time') {
+            len = Math.ceil((dataExtent[1] - dataExtent[0]) / (this.scale as TimeScale)._approxInterval);
```

Comment:
From the engineering point of view, it may not be a good idea to check the type of scale in Axis. I didn't dive into the code but does this change means the `getExtent` of time axis returns a result that is not expected? If so, should we change the logic inside time scale? For example, does other functions have bug with time axis when calling `getExtent`?

Reply:
> does this change means the getExtent of time axis returns a result that is not expected?

No, `getExtent` returns perfectly. Nothing wrong about that.  The problem here is `dataExtent` could be extremely larger that `axisExtent` since `time dataExtent` are timestamps. 

For example in this case: `505854800.9370117`

We should treat `time axis` as `category axis`  to calcule how many section on axis. @Ovilia